src/components/Modals/PrismicDocumentModal/PrismicDocumentModal.js

```diff
@@ -55,9 +56,18 @@ const PrismicDocumentModal = ({ prismicDocumentId }: Props) => {
 
   React.useEffect(() => {
     async function fetchPrismicData() {
-      const fetchPrismicHTMLResponse = await Prismic.queryDocumentsByID(prismicDocumentId);
-      setDocumentHTMLData(fetchPrismicHTMLResponse);
-      setIsPrismicHTMLFetched(true);
+      try {
+        const prismicDocument = await Prismic.queryDocumentsByID(prismicDocumentId);
+        const prismicContent = [];
+        mapFromDocumentDataToString(prismicDocument?.title, prismicContent, true);
+        mapFromDocumentDataToString(prismicDocument?.subtitle, prismicContent, true);
+        mapFromDocumentDataToString(prismicDocument?.content, prismicContent, true);
+        const prismicHTMLResponse = prismicContent.join('');
+        setDocumentHTMLData(prismicHTMLResponse);
+        setIsPrismicHTMLFetched(true);
+      } catch (error) {
+        reportErrorLog('Prismic content fetch failed', { error, documentId: prismicDocumentId });
```

Comment:
Shouldn't `setIsPrismicHTMLFetched` be called on error as well just to not show infinite spinner? Maybe Toast or error/warning message can be rendered in UI as well? wdyt?

Reply:
@poocart Done
![errorMessage](https://user-images.githubusercontent.com/66996635/124747913-19e0dc80-df40-11eb-87d4-1a589634f0aa.png)